Apps/Common/SIMSolver.h

```diff
@@ -195,12 +196,36 @@ template<class T1> class SIMSolver : public SIMadmin
     return restartStep;
   }
 
+  //! \brief Handles application data output.
```

Comment:
You're right there. No need to return the pointer as it is not used (and shouldn't be) outside. Just a leftover from the SIM:: function. Will take away.